Split up a couple expressions to work around slow type checking.

rudkx opened the pull request:
The type checker is particularly slow on these expressions. I've opened
`https://bugs.swift.org/browse/SR-5322` to investigate the type checker
issue, but in the meantime it would be nice to speed up builds. This
cuts more than a minute from my unoptimized asserts-enabled build.

rudkx commented:
@swift-ci Please smoke test

rudkx commented:
@parkera Is this okay to merge?

parkera commented:
Can we put a SR or radar number in the comment so we know when it's ok to remove this hack?

rudkx commented:
Sure - I've added a commit that mentions the SR number, and updated the SR to point back to this PR so that we know to revert it once the type checker performance issue is resolved.

rudkx commented:
@swift-ci Please smoke test

phausler commented:
@swift-ci please test and merge
